specification/machinelearning/resource-manager/Microsoft.MachineLearning/2016-04-01/swagger/workspaces.json

```diff
@@ -0,0 +1,704 @@
+          "type": "string",
+          "description": "error message"
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "Workspace": {
+      "type": "object",
+      "description": "An object that represents a machine learning workspace.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/WorkspaceProperties",
+          "description": "The properties of the machine learning workspace.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WorkspaceProperties": {
+      "type": "object",
+      "description": "The properties of a machine learning workspace.",
+      "required": [
+        "userStorageAccountId"
+      ],
+      "properties": {
+        "userStorageAccountId": {
+          "description": "The fully qualified arm id of the storage account associated with this workspace.",
+          "type": "string"
+        },
+        "ownerEmail": {
+          "description": "The email id of the owner for this workspace.",
+          "type": "string"
+        },
+        "workspaceType": {
+          "type": "string",
+          "enum": [
+            "Production",
+            "Free",
+            "Anonymous",
+            "PaidStandard",
+            "PaidPremium"
+          ],
+          "x-ms-enum": {
+            "name": "WorkspaceType",
+            "modelAsString": false
+          },
+          "readOnly": true,
+          "description": "The type of this workspace."
+        },
+        "workspaceState": {
+          "type": "string",
+          "enum": [
+            "Deleted",
+            "Enabled",
+            "Disabled",
+            "Migrated",
+            "Updated",
+            "Registered",
+            "Unregistered"
+          ],
+          "x-ms-enum": {
+            "name": "WorkspaceState",
+            "modelAsString": false
+          },
+          "readOnly": true,
+          "description": "The current state of workspace resource."
+        },
+        "workspaceId": {
+          "description": "The immutable id associated with this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationTime": {
+          "description": "The creation time for this workspace resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "workspaceAuthorizationToken": {
+          "description": "The authorization token to access this workspace. This field will only be populated when the response is returned from a PUT/PATCH request.",
+          "type": "string",
+          "readOnly": true
+        },
+        "studioEndpoint": {
+          "description": "The regional endpoint for the machine learning studio service which hosts this workspace.",
+          "type": "string",
+          "readOnly": true
+        },
+        "keyVaultIdentifierId": {
```

Comment:
Is this the ARM resource ID of the keyvault resource?

Reply:
Nope... This is the url of the key (in Azure key vault) which we use to encrypt workspace